processing/src/main/java/org/apache/carbondata/processing/merger/RowResultMergerProcessor.java

```diff
@@ -57,8 +62,10 @@
 
   public RowResultMergerProcessor(String databaseName,
       String tableName, SegmentProperties segProp, String[] tempStoreLocation,
-      CarbonLoadModel loadModel, CompactionType compactionType) {
+      CarbonLoadModel loadModel, CompactionType compactionType, List<String> partitionNames) {
     this.segprop = segProp;
+    this.partitionNames = partitionNames;
+    this.loadModel = loadModel;
```

Comment:
How about keeping the segment path in this class and use it in line 162 instead of keeping the whole load model in this class?

Reply:
But it needs task number as well, thats why loadModel kept at class level instead of keeping two parameters